### PR TITLE
Update elasticsearch to latest version

### PR DESCRIPTION
**Pull request by d34dh0r53**
This updates elasticsearch to it latest version to the current version for the Mitaka work. This also removes the site plugins that we were using in the legacy stack as they have been deprecated in this version of elasticsearch. I've removed the mappings file as it was causing errors and the new version of elasticsearch is doing a good job of dynamically configuring the mappings.

Connects rcbops/u-suk-dev#176

### CODE REVIEW

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by Apsu** [comment]
RPC-AIO-Ceph-Liberty-PR: Fail

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by d34dh0r53** [comment]
recheck_swift
Compute failure

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml [hunk 1]
```diff
@@ -17,8 +17,8 @@
   template:
     src: "{{ item }}"
     dest: "/etc/elasticsearch/{{ item }}"
-    owner: "root"
-    group: "root"
+    owner: "elasticsearch"
```
Comment (sigmavirus24) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: Should we convert these to role defaults too?

**Comment by sigmavirus24** [comment]
One question, otherwise, LGTM.

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml (hunk 1 above)
Comment (prometheanfire) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: yes, though given we only consume here...

**Comment by andymcc** [comment]
Greenfields works ok - are we concerned about upgrades at all? i believe those won't work due to plugins (which we can remove in the upgrade script easily enough), and (potentially more tricky) there are incorrect logstash filters that prevent elasticsearch from starting because of type mismatch between ip and string (I believe this is fields that are labeled as both ip and string in different places). The second one may require a crafty solution wrt dumping/reimporting the data.

That's probably ok - we have to sort out the filters etc anyway - but as with the kibana PR my concern is that we are relying on future tasks that HAVE to happen in order for this to work properly - we don't really have any gate checks that test any functionality so that these pass the gate is mostly irrelevant (they deploy fine).

Otherwise this LGTM.

**Comment by sigmavirus24** [comment]
@andymcc this is the work for greenfields only. When we start working on 13.1 we'll start bothering about upgrades. :)

**Comment by claco** [comment]
Question: How do we not lose track of this additional work in 13.1? Do we ensure we have a card on the board now for the milestone to 'review' this area of code?

**Comment by sigmavirus24** [comment]
There were rcbops/u-suk-dev#181 and rcbops/u-suk-dev#182 and I just added rcbops/u-suk-dev#237 to cover this specifically.

**Comment by d34dh0r53** [comment]
recheck_swift

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by prometheanfire** [comment]
I'm :+1: on this

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml (hunk 1 above)
Comment (mancdaz) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: For now.....but the correct pattern as @sigmavirus24 suggested is to convert to role defaults.

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml [hunk 2]
```diff
@@ -42,50 +42,6 @@
...
-  command: ./plugin -install {{ item }}
-  args:
-    chdir: /usr/share/elasticsearch/bin
-    creates: /usr/share/elasticsearch/plugins/bigdesk
-  with_items:
-    - lukas-vlcek/bigdesk/2.4.0
-  tags:
-    - elasticsearch-plugins
-    - elasticsearch-install
-
```
Comment (mancdaz) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml]: how come we don't need any of these plugins now?

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml [hunk 3]
```diff
@@ -48,21 +48,6 @@
...
-    group: "root"
-    mode: "0755"
-  tags:
-    - elasticsearch-post-install
-
-- name: Run ElasticSearch schema mapping script
-  shell: /opt/mapping.sh
-  tags:
-    - elasticsearch-post-install
-
```
Comment (mancdaz) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: why is this not needed now?

**Comment by mancdaz** [comment]
The commit message needs a bit more detail explaining why we're removing a bunch of the install steps. I'm sure there's a perfectly good reason as to why they aren't needed in the new version of elasticsearch, but we need to explain in the commit.

**Comment by d34dh0r53** [comment]
@mancdaz I have updated the commit message and the initial comment to reflect the fact that site plugins are deprecated in this version of elasticsearch.  The reason for the mapping removal is that the new version of elasticsearch seems to do a really good job of type mapping, but we may still need to add some custom mappings once the filter work begins.

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml (hunk 3 above)
Comment (d34dh0r53) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: This mapping file actually causes elasticsearch to not start, and it seems that the new version of elasticsearch does a really good job of dynamic mapping.  We may need to add some custom mappings once the filter work begins but for now it is not needed.

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml (hunk 2 above)
Comment (d34dh0r53) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml]: Site plugins are deprecated in `2.3.x` and will be removed in the next major.

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by d34dh0r53** [comment]
recheck_swift

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml (hunk 2 above)
Comment (evrardjp) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_install.yml]: Are the features not needed anymore or included in the default ES?

File: rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml (hunk 1 above)
Comment (evrardjp) [inline rpcd/playbooks/roles/elasticsearch/tasks/elasticsearch_post_install.yml]: Agreed

**Comment by evrardjp** [comment]
I think the commit message still need to explain the why - and how do we ensure feature continuity.

Except that, I'm good, but I haven't tested it.

**Comment by d34dh0r53** [comment]
Commit message updated.

**Comment by Apsu** [comment]
RPC-AIO-Liberty-PR: Fail

**Comment by sigmavirus24** [comment]
So I have an AIO with this, #1245 and #1246 running and it seems to be working. Logstash is shipping things to elasticsearch and kibana is accessible and showing the items in queries out of elasticsearch.

So a 👍 having tested everything.

**Comment by sigmavirus24** [comment]
recheck_swift

**Comment by andymcc** [comment]
replicated what @sigmavirus24 has seen and this works with #1084 and #1246.
👍

**Pull request merged**